net/netfilter/ip6t_sockopt.c

```diff
@@ -0,0 +1,444 @@
+
+static int get_info(FAR struct ip6t_getinfo *get, FAR socklen_t *len)
+{
+  FAR struct ip6t_replace *repl;
+
+  if (*len != sizeof(*get))
+    {
+      return -EINVAL;
+    }
+
+  repl = ip6t_table_repl(get->name);
+  if (repl == NULL)
+    {
+      return -ENOENT;
+    }
+
+  get->valid_hooks = repl->valid_hooks;
+  memcpy(get->hook_entry, repl->hook_entry, sizeof(get->hook_entry));
+  memcpy(get->underflow, repl->underflow, sizeof(get->underflow));
+  get->num_entries = repl->num_entries;
+  get->size = repl->size;
+
+  return OK;
+}
+
+/****************************************************************************
+ * Name: get_entries
+ *
+ * Description:
+ *   Fill entry info into ip6t_get_entries structure.
+ *
+ * Input Parameters:
+ *   get, len - The parameters from getsockopt.
+ *
+ ****************************************************************************/
+
+static int get_entries(FAR struct ip6t_get_entries *get, FAR socklen_t *len)
+{
+  FAR struct ip6t_replace *repl;
+
+  if (*len < sizeof(*get) || *len != sizeof(*get) + get->size)
+    {
+      return -EINVAL;
+    }
+
+  repl = ip6t_table_repl(get->name);
+  if (repl == NULL)
+    {
+      return -ENOENT;
+    }
+
+  if (get->size != repl->size)
+    {
+      return -EAGAIN;
+    }
+
+  memcpy(get->entrytable, repl->entries, get->size);
+
+  return OK;
+}
+
+/****************************************************************************
+ * Name: check_replace
+ *
+ * Description:
+ *   Check whether an ip6t_replace structure from user space is valid.
+ *
+ * Input Parameters:
+ *   repl - The ip6t_replace structure to check.
+ *
+ * Returned Value:
+ *   OK if repl is valid, otherwise -EINVAL.
+ *
+ ****************************************************************************/
+
+static int check_replace(FAR const struct ip6t_replace *repl)
+{
+  FAR struct ip6t_entry *entry;
+  unsigned int entry_count = 0;
+
+  ip6t_entry_for_every(entry, repl->entries, repl->size)
+    {
+      entry_count++;
+      if (entry->next_offset == 0)
+        {
+          return -EINVAL;
+        }
+    }
+
+  if (entry_count != repl->num_entries)
+    {
+      return -EINVAL;
+    }
+
+  /* May add more checks later. */
+
+  return OK;
+}
+
+/****************************************************************************
+ * Name: replace_entries
+ *
+ * Description:
+ *   Apply replace data to kernel tables.
+ *
+ * Input Parameters:
+ *   repl, len - The parameters from setsockopt.
+ *
+ ****************************************************************************/
+
+static int replace_entries(FAR const struct ip6t_replace *repl,
+                           socklen_t len)
+{
+  int ret;
+  FAR struct ip6t_replace *new_repl;
+  FAR struct ip6t_table_s *table = ip6t_table(repl->name);
+
+  if (table == NULL || table->repl == NULL)
+    {
+      return -ENOENT;
+    }
+
+  if (len < sizeof(*repl) || len != sizeof(*repl) + repl->size ||
```

Comment:
```suggestion
  if (len != sizeof(*repl) + repl->size ||
```

Reply:
If `len < sizeof(*repl)`, it's dangerous to access `repl->size`, so I check `len < sizeof(*repl)` first.